dex/src/main/java/io/crate/data/CollectingBatchIterator.java

```diff
@@ -89,13 +93,15 @@ private CollectingBatchIterator(Supplier<CompletableFuture<? extends Iterable<?
         return newInstance(
             source::close,
             source::kill,
-            () -> processSource.apply(source).whenComplete((r, f) -> source.close()));
+            () -> processSource.apply(source).whenComplete((r, f) -> source.close()),
+            false);
```

Comment:
Should we use `source.involesIO` here? Or even let the API callee decide, in case that the function itself involves IO?